docs/proposals/separate-config.md

```diff
@@ -0,0 +1,32 @@
+We should separate the config for our separate processes.  We have
+processes for kube-apiserver, kube-controller-manager, kube-scheduler, openshift-apiserver,
+and openshift-controller manager.  All of these are currently fed based on master-config.yaml,
+but they have many discrete, non-overlapping values.
+
+Coarse separation:
+ 1. openshift-apiserver - we will have a config for this for the forseeable future.
+ It will be different than a kube-apiserver config.
+ We should start this struct now.
+ 2. openshift-controllers - we will have a config for this for the forseeable future.
+ It will be different than a kube-controller-manager config.
+ We should start this struct now.
+ 3. kube-scheduler - We should not be special for this.
+ We have no need for custom flags.
+ We should use the flags as indicated by upstream so that we can take advantage of any migration they provide.
```

Comment:
flags are deprecated. The component file is to be used. Do we expose that to users?

Reply:
> flags are deprecated. The component file is to be used. Do we expose that to users?

My point in writing this up is that I do not think that we should invent a file format for it.  Today we use flags like all old consumers.  I think it is up to @aveshagarwal and @openshift/sig-pod to decide how they'd like to configure it long term.  I'd suggest following upstream guidance instead of providing an openshift wrapper.